### ISSUE TITLE
RestEqlSearchAction trips assertion and tries to send a response to a REST channel twice

### ISSUE THREAD

**Issue by original-brownbear** [elastic/elasticsearch#89932]
I see this failure in cloud logs:

```
[2022-09-02T11:38:33,711][ERROR][org.elasticsearch.xpack.eql.plugin.RestEqlSearchAction] [instance-0000000049] failed to send failure response
java.lang.IllegalStateException: Channel is already closed
	at org.elasticsearch.rest.RestController$ResourceHandlingHttpChannel.close(RestController.java:669) ~[elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.rest.RestController$ResourceHandlingHttpChannel.sendResponse(RestController.java:662) ~[elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.xpack.eql.plugin.RestEqlSearchAction$1.onFailure(RestEqlSearchAction.java:94) [x-pack-eql-8.3.3.jar:8.3.3]
	at org.elasticsearch.rest.action.RestCancellableNodeClient$1.onFailure(RestCancellableNodeClient.java:96) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.client.internal.node.NodeClient$ActionResponseTaskListener.onFailure(NodeClient.java:185) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.tasks.TaskManager$1.onFailure(TaskManager.java:180) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.support.ContextPreservingActionListener.onFailure(ContextPreservingActionListener.java:38) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.ActionListener$Delegating.onFailure(ActionListener.java:92) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.xpack.ql.plugin.TransportActionUtils.lambda$executeRequestWithRetryAttempt$0(TransportActionUtils.java:71) [x-pack-ql-8.3.3.jar:8.3.3]
	at org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.xpack.eql.execution.PlanExecutor.lambda$eql$0(PlanExecutor.java:80) [x-pack-eql-8.3.3.jar:8.3.3]
	at org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.ActionListener$Delegating.onFailure(ActionListener.java:92) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.client.internal.node.NodeClient$ActionResponseTaskListener.onFailure(NodeClient.java:185) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.tasks.TaskManager$1.onFailure(TaskManager.java:180) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:53) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.tasks.TaskManager.registerAndExecute(TaskManager.java:165) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.client.internal.node.NodeClient.executeLocally(NodeClient.java:113) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.client.internal.node.NodeClient.doExecute(NodeClient.java:91) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.client.internal.support.AbstractClient.execute(AbstractClient.java:380) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.client.internal.FilterClient.doExecute(FilterClient.java:57) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.client.internal.ParentTaskAssigningClient.doExecute(ParentTaskAssigningClient.java:55) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.client.internal.support.AbstractClient.execute(AbstractClient.java:380) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.xpack.eql.execution.search.PITAwareQueryClient.close(PITAwareQueryClient.java:142) [x-pack-eql-8.3.3.jar:8.3.3]
	at org.elasticsearch.xpack.eql.execution.sequence.TumblingWindow.lambda$execute$1(TumblingWindow.java:129) [x-pack-eql-8.3.3.jar:8.3.3]
	at org.elasticsearch.action.ActionListener$RunAfterActionListener.onFailure(ActionListener.java:378) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.client.internal.node.NodeClient$ActionResponseTaskListener.onFailure(NodeClient.java:185) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.tasks.TaskManager$1.onFailure(TaskManager.java:180) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.support.ContextPreservingActionListener.onFailure(ContextPreservingActionListener.java:38) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.ActionListener$Delegating.onFailure(ActionListener.java:92) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.ActionListener$Delegating.onFailure(ActionListener.java:92) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.ActionListener$Delegating.onFailure(ActionListener.java:92) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.ActionListener$RunAfterActionListener.onFailure(ActionListener.java:376) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.search.AbstractSearchAsyncAction.raisePhaseFailure(AbstractSearchAsyncAction.java:752) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseFailure(AbstractSearchAsyncAction.java:729) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:433) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseDone(AbstractSearchAsyncAction.java:761) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.search.AbstractSearchAsyncAction.onShardFailure(AbstractSearchAsyncAction.java:513) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.search.AbstractSearchAsyncAction$1.onFailure(AbstractSearchAsyncAction.java:350) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.ActionListenerResponseHandler.handleException(ActionListenerResponseHandler.java:48) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.transport.TransportService.handleSendRequestException(TransportService.java:759) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:742) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.transport.TransportService.sendChildRequest(TransportService.java:812) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.transport.TransportService.sendChildRequest(TransportService.java:800) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.search.TransportOpenPointInTimeAction.lambda$doExecute$0(TransportOpenPointInTimeAction.java:88) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.search.TransportSearchAction$1$1.executePhaseOnShard(TransportSearchAction.java:339) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.search.AbstractSearchAsyncAction.lambda$performPhaseOnShard$4(AbstractSearchAsyncAction.java:335) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.action.search.AbstractSearchAsyncAction$2.doRun(AbstractSearchAsyncAction.java:395) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:769) [elasticsearch-8.3.3.jar:?]
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26) [elasticsearch-8.3.3.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
	Suppressed: org.elasticsearch.action.ActionRequestValidationException: Validation Failed: 1: id is empty;
		at org.elasticsearch.action.ValidateActions.addValidationError(ValidateActions.java:15) ~[elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.action.search.ClosePointInTimeRequest.validate(ClosePointInTimeRequest.java:45) ~[elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:45) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.tasks.TaskManager.registerAndExecute(TaskManager.java:165) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.client.internal.node.NodeClient.executeLocally(NodeClient.java:113) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.client.internal.node.NodeClient.doExecute(NodeClient.java:91) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.client.internal.support.AbstractClient.execute(AbstractClient.java:380) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.client.internal.FilterClient.doExecute(FilterClient.java:57) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.client.internal.ParentTaskAssigningClient.doExecute(ParentTaskAssigningClient.java:55) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.client.internal.support.AbstractClient.execute(AbstractClient.java:380) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.xpack.eql.execution.search.PITAwareQueryClient.close(PITAwareQueryClient.java:142) [x-pack-eql-8.3.3.jar:8.3.3]
		at org.elasticsearch.xpack.eql.execution.sequence.TumblingWindow.lambda$execute$1(TumblingWindow.java:129) [x-pack-eql-8.3.3.jar:8.3.3]
		at org.elasticsearch.action.ActionListener$RunAfterActionListener.onFailure(ActionListener.java:378) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.client.internal.node.NodeClient$ActionResponseTaskListener.onFailure(NodeClient.java:185) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.tasks.TaskManager$1.onFailure(TaskManager.java:180) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.action.support.ContextPreservingActionListener.onFailure(ContextPreservingActionListener.java:38) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.action.ActionListener$Delegating.onFailure(ActionListener.java:92) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.action.ActionListener$Delegating.onFailure(ActionListener.java:92) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.action.ActionListener$Delegating.onFailure(ActionListener.java:92) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.action.ActionListener$RunAfterActionListener.onFailure(ActionListener.java:376) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.action.search.AbstractSearchAsyncAction.raisePhaseFailure(AbstractSearchAsyncAction.java:752) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseFailure(AbstractSearchAsyncAction.java:729) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:433) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseDone(AbstractSearchAsyncAction.java:761) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.action.search.AbstractSearchAsyncAction.onShardFailure(AbstractSearchAsyncAction.java:513) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.action.search.AbstractSearchAsyncAction$1.onFailure(AbstractSearchAsyncAction.java:350) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.action.ActionListenerResponseHandler.handleException(ActionListenerResponseHandler.java:48) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.transport.TransportService.handleSendRequestException(TransportService.java:759) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:742) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.transport.TransportService.sendChildRequest(TransportService.java:812) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.transport.TransportService.sendChildRequest(TransportService.java:800) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.action.search.TransportOpenPointInTimeAction.lambda$doExecute$0(TransportOpenPointInTimeAction.java:88) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.action.search.TransportSearchAction$1$1.executePhaseOnShard(TransportSearchAction.java:339) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.action.search.AbstractSearchAsyncAction.lambda$performPhaseOnShard$4(AbstractSearchAsyncAction.java:335) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.action.search.AbstractSearchAsyncAction$2.doRun(AbstractSearchAsyncAction.java:395) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:769) [elasticsearch-8.3.3.jar:?]
		at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26) [elasticsearch-8.3.3.jar:?]
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
		at java.lang.Thread.run(Thread.java:833) [?:?]
```

prior to #89873 this would lead to a memory leak. Also this would trip an assertion in `org.elasticsearch.action.support.TransportAction#execute` but the above is from a production cluster.
It seems we have to fix something about the request instantiation or the validation in the rest action here to fix this.

**Comment by luigidellaquila** [comment]
Hi @original-brownbear 

I think this was already fixed in v 8.4, see https://github.com/elastic/elasticsearch/pull/87498

**Comment by original-brownbear** [comment]
ah nice thanks @luigidellaquila :)
